mailpoet/assets/js/src/common/premium_modal/index.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+  ComponentProps,
+  EventHandler,
+  FocusEvent,
+  KeyboardEvent,
+  MouseEvent,
+} from 'react';
+import { Button, Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+type Props = Omit<ComponentProps<typeof Modal>, 'title' | 'onRequestClose'> & {
+  // Fix type from "@types/wordpress__components" where it is defined as a union of event
+  // handlers, resulting in a function requiring intersection of all of the event types.
+  onRequestClose: EventHandler<KeyboardEvent | MouseEvent | FocusEvent>;
+};
+
+export function PremiumModal({ children, ...props }: Props): JSX.Element {
```

Comment:
You do not use the `children` here.

Reply:
Ahh, yeah, another consequence of untangling the rabbit hole in two separate tickets. In https://github.com/mailpoet/mailpoet/pull/4312 it's used.